include/convex_bodies/hpolytope.h

```diff
@@ -100,6 +112,7 @@ class HPolytope {
     void set_InnerBall(std::pair<Point,NT> const& innerball) //const
     {
         _inner_ball = innerball;
+        has_ball = true;
     }
 
     void set_interior_point(Point const& r)
```

Comment:
Can you add a TODO here to fix this function?
Because when we change the center of the inner ball we also need to change its radius.

Reply:
here we set not just the center, but also the radius, since innerball is a pair of center,radius; I decided to set the has_ball to true here since I assumed that whenever this function might be used, it would be given a valid innerball, but tbh I haven't seen anywhere in the code where this function is called anyways.

Reply:
Fisrt, we need to check that the point is indeed inside the polytope and only then set it as center of the inner ball. Then, we need to compute the distances of the center from all the facets and take the minimum. The latter distance is the radius of the maximum inscribed ball centered at the given point.

The distance between a point p and the i-th facet is given by: (b_i - a_i^Tp) / ||a_i||.
Check also the function `get_dists()` below; it returns the distance from the origin to each facet.